sale_blanket_order/wizard/create_sale_orders.py

```diff
@@ -0,0 +1,205 @@
+# Copyright 2018 ACSONE SA/NV
+# Copyright 2019 Elico Corp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models, api, _
+from odoo.tools import float_is_zero
+from collections import defaultdict
+from odoo.exceptions import UserError
+
+
+class BlanketOrderWizard(models.TransientModel):
+    _name = 'sale.blanket.order.wizard'
+    _description = 'Blanket Order Wizard'
+
+    @api.model
+    def _default_order(self):
+        # in case the cron hasn't run
+        self.env['sale.blanket.order'].expire_orders()
+        if not self.env.context.get('active_id'):
+            return False
+        blanket_order = self.env['sale.blanket.order'].search(
+            [('id', '=', self.env.context['active_id'])], limit=1)
+        if blanket_order.state == 'expired':
+            raise UserError(_('You can\'t create a sale order from '
+                              'an expired blanket order!'))
+        return blanket_order
+
+    @api.model
+    def _check_valid_blanket_order_line(self, bo_lines):
+        precision = self.env['decimal.precision'].precision_get(
+            'Product Unit of Measure')
+        company_id = False
+
+        for line in bo_lines:
+
+            if line.order_id.state != 'open':
+                raise UserError(
+                    _('Sale Blanket Order %s is not open') %
+                    line.order_id.name)
+
+            if float_is_zero(
+                    line.remaining_uom_qty, precision_digits=precision):
+                raise UserError(
+                    _('The sale has already been completed.'))
+
+            line_company_id = line.company_id and line.company_id.id or False
+            if company_id is not False \
+                    and line_company_id != company_id:
+                raise UserError(
+                    _('You have to select lines '
+                      'from the same company.'))
+            else:
+                company_id = line_company_id
+
+    @api.model
+    def _default_lines(self):
+        blanket_order_line_obj = self.env['sale.blanket.order.line']
+        blanket_order_line_ids = self.env.context.get('active_ids', False)
+        active_model = self.env.context.get('active_model', False)
+
+        if active_model == 'sale.blanket.order':
+            bo_lines = self._default_order().line_ids
+        else:
+            bo_lines = blanket_order_line_obj.browse(blanket_order_line_ids)
+
+        self._check_valid_blanket_order_line(bo_lines)
+
+        lines = [(0, 0, {
+            'blanket_line_id': l.id,
+            'product_id': l.product_id.id,
+            'date_schedule': l.date_schedule,
+            'remaining_uom_qty': l.remaining_uom_qty,
+            'price_unit': l.price_unit,
+            'product_uom': l.product_uom,
+            'qty': l.remaining_uom_qty,
+            'partner_id': l.partner_id,
+        }) for l in bo_lines]
+        return lines
+
+    blanket_order_id = fields.Many2one(
```

Comment:
When are `blanket_order_id` and `sale_order_id` filled? Is there perhaps a `default=_default_order` needed somewhere?